plugins/discovery-gce/src/main/java/org/elasticsearch/cloud/gce/GceComputeService.java

```diff
@@ -21,21 +21,54 @@
 
 import com.google.api.services.compute.model.Instance;
 import org.elasticsearch.common.component.LifecycleComponent;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.unit.TimeValue;
 
+import java.util.Arrays;
 import java.io.IOException;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
 
 public interface GceComputeService extends LifecycleComponent<GceComputeService> {
-    final class Fields {
-        public static final String PROJECT = "cloud.gce.project_id";
-        public static final String ZONE = "cloud.gce.zone";
-        public static final String REFRESH = "cloud.gce.refresh_interval";
-        public static final String TAGS = "discovery.gce.tags";
-        public static final String VERSION = "Elasticsearch/GceCloud/1.0";
-
-        public static final String RETRY = "cloud.gce.retry";
-        public static final String MAXWAIT = "cloud.gce.max_wait";
-    }
+
+    /**
+     * GCE API Version: Elasticsearch/GceCloud/1.0
+     */
+    String VERSION = "Elasticsearch/GceCloud/1.0";
+
+    // cloud.gce settings
+
+    /**
+     * cloud.gce.project_id: Google project id
+     */
+    Setting<String> PROJECT_SETTING = Setting.simpleString("cloud.gce.project_id", false, Setting.Scope.CLUSTER);
+
+    /**
+     * cloud.gce.zone: Google Compute Engine zones
+     */
+    Setting<List<String>> ZONE_SETTING =
+        Setting.listSetting("cloud.gce.zone", Collections.emptyList(), s -> s, false, Setting.Scope.CLUSTER);
+
+    /**
+     * cloud.gce.refresh_interval: How long the list of hosts is cached to prevent further requests to the AWS API. 0 disables caching.
+     * A negative value will cause infinite caching. Defaults to 0s.
+     */
+    Setting<TimeValue> REFRESH_SETTING =
+        Setting.timeSetting("cloud.gce.refresh_interval", TimeValue.timeValueSeconds(0), false, Setting.Scope.CLUSTER);
+
+    /**
+     * cloud.gce.retry: Should we retry calling GCE API in case of error? Defaults to true.
+     */
+    Setting<Boolean> RETRY_SETTING = Setting.boolSetting("cloud.gce.retry", true, false, Setting.Scope.CLUSTER);
+
+    /**
+     * cloud.gce.max_wait: How long exponential backoff should retry.
```

Comment:
It's not entirely clear from the Javadoc whether this specifies the total time interval that is spent in retry or at the last attempt of retry or something else.